synapse/util/logcontext.py

```diff
@@ -312,10 +314,12 @@ def copy_to(self, record):
         another LoggingContext
         """
 
-        # 'request' is the only field we currently use in the logger, so that's
-        # all we need to copy
+        # we track the current request
         record.request = self.request
 
+        # we also track the current scope:
+        record.scope = self.scope
```

Comment:
Do we need to copy this to child log contexts?

Reply:
Yes, If not: a logcontext created as a child of another loose us access to the current scope. A less statefull way would involve doing a walk up the logcontexts to find an ancestor logcontext with an active scope. I would prefer to do this since it results in a little cleaner state management but I presume the log contexts can get a little deep?